src/mainstay.rs

```diff
@@ -0,0 +1,92 @@
+use reqwest;
+use crate::config::Mainstay;
+use nostr::Event;
+use serde_json::json;
+use serde::{Serialize, Deserialize};
+use std::collections::HashMap;
+use base64::encode;
+use sha2::{Digest, Sha256};
+use hex;
+
+pub struct Request(reqwest::RequestBuilder);
+
+#[derive(Serialize, Deserialize)]
+pub struct Payload {
+    pub commitment: String,
+    pub position: u64,
+    pub token: String,
+}
+
+impl Request {
```

Comment:
If we can integrate this file in `src/services/notaryd.rs` here we’ll live all the attestation logic.

Reply:
This makes sense - just trying to understand how `notaryd` interacts with the events generated on `civkitd` to be committed. They are received from `CivkitServiceClient`. Also will `notaryd` have DB access?